### DISCUSSION TITLE
An unknown Strapi error was returned while calling Route http://127.0.0.1:8000/test/2

### DISCUSSION THREAD

**Discussion by bast111** [dbfx/laravel-strapi#15] [Q&A]
Hi guys, being fairly new to Laravel i wanted to start implement a strapi backend following that tutorial [here](https://strapi.io/blog/strapi-and-laravel)

All worked fine at first, I created a route to fetch all my localites
```
Route::get('/test', function () {
    $strapi = new LaravelStrapi();
    return $localites = $strapi->collection('api/localites');
});
```
![image](https://user-images.githubusercontent.com/31862563/161993527-86e7e56c-e112-4a17-bfc1-4772b03149c9.png)

So I kept going and tried to create the route for one item 
```
//Route pour 1 localite
Route::get('/test/{id}', function ($id) {
    $strapi = new LaravelStrapi();
    return $localites = $strapi->entry('api/localites', $id);
});
```
And this is what I get : 
![image](https://user-images.githubusercontent.com/31862563/161994382-8fcb995f-3861-4ae4-a49b-e280d1c09259.png)

I can see that the api has been called successfully but I don't understand what is the problem
![image](https://user-images.githubusercontent.com/31862563/161994741-870e5e5e-3c24-4888-bbeb-b90374b18115.png)

If anyonce can give me some insights would be really appreciated :) Thanks

**Comment by dbfx** [comment]
Hmm strange. On line 95 of LaravelStrapi.php you'll see this if: 

```
        if (!isset($entry['id'])) {
```

Maybe in that dd($entry) and see what you are getting back from Strapi?

**Reply by bast111** [reply]
Got me this with the dd
![image](https://user-images.githubusercontent.com/31862563/161996637-a9d7c632-fc36-45ce-99a2-0d2ac00d036c.png)

Strapi still shows good request 
![image](https://user-images.githubusercontent.com/31862563/161996950-96ee45e9-9775-4656-9cf6-4d8198117082.png)
![image](https://user-images.githubusercontent.com/31862563/161997880-7236948e-c552-434a-b071-24490d23755f.png)

**Reply by dbfx** [reply]
Hmm can you try without "api/" in your $type

**Reply by bast111** [reply]
Okay so here's without the "api/"
```
//Route pour collection complète des localité    
Route::get('/test', function () {
    $strapi = new LaravelStrapi();
    return $localites = $strapi->collection('localites');
});

//Route pour 1 localite
Route::get('/test/{id}', function ($id) {
    $strapi = new LaravelStrapi();
    return $localites = $strapi->entry('localites', $id);
});
```
Collection Route (not working anymore)
![image](https://user-images.githubusercontent.com/31862563/161999330-7abecf2e-5adb-48ff-b5d5-9ea30fae20fe.png)
Entry Route (still not working)
![image](https://user-images.githubusercontent.com/31862563/161999416-7f2828ec-06c3-4acc-aefb-fbaa1dcb6c09.png)
And Strapi console (404 request)
![image](https://user-images.githubusercontent.com/31862563/161999604-33746bbd-5f98-480f-ba1d-ad36d7d7e5a5.png)


It's weird that i need to add this to allow him to fetch properly but idk didn't manage to make it work without it
But it's the same with the Strapi URL i have to add asw the api/
![image](https://user-images.githubusercontent.com/31862563/161999914-efcbcb64-5702-4eee-a933-fe41572edf06.png)

Maybe i did something wrong on the strapi side ? But I juste created normal content types 
![image](https://user-images.githubusercontent.com/31862563/162000441-41caf22c-a3db-4274-be29-4d4e03091b8e.png)

**Reply by bast111** [reply]
In Strapi v4 docs it is specified that the url content /api/
![image](https://user-images.githubusercontent.com/31862563/162007609-0cc834d5-1bb1-4cc8-81d9-4b434fda35bb.png)

So everything looks good on that side but i don't see anywhere in the LaravelStrapi.php collection and entry function adding api/ to the request url so i thought i would add it myself in the $type name. It seemed working better than adding it into SRAPI_URL in the env file

**Comment by bast111** [comment]
hi guys, still struggling with the same problem.. any lead ?

**Comment by opryshok** [comment]
Hi, I have the same problem with Laravel 9. Any ideas?
